assets/js/modules/analytics-4/components/widgets/LeastEngagingPagesWidget.js

```diff
@@ -0,0 +1,240 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import {
+	DATE_RANGE_OFFSET,
+	MODULES_ANALYTICS_4,
+} from '../../datastore/constants';
+import {
+	MetricTileTable,
+	MetricTileTablePlainText,
+} from '../../../../components/KeyMetrics';
+import Link from '../../../../components/Link';
+import { ZeroDataMessage } from '../../../analytics/components/common';
+import { numFmt } from '../../../../util';
+import whenActive from '../../../../util/when-active';
+import ConnectGA4CTATileWidget from './ConnectGA4CTATileWidget';
+import useViewOnly from '../../../../hooks/useViewOnly';
+const { useSelect, useInViewSelect } = Data;
+
+function LeastEngagingPagesWidget( props ) {
+	const { Widget } = props;
+
+	const viewOnlyDashboard = useViewOnly();
+
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+		} )
+	);
+
+	const pageViewsReportOptions = {
+		...dates,
+		dimensions: [ 'pagePath' ],
+		metrics: [ { name: 'screenPageViews' } ],
+		orderby: [
+			{
+				metric: { metricName: 'screenPageViews' },
+				desc: true,
+			},
+		],
+	};
+
+	const pageViewsReport = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( pageViewsReportOptions )
+	);
+
+	const medianIndex = parseInt( pageViewsReport?.rowCount / 2, 10 );
+	const medianPageViews =
+		parseInt(
+			pageViewsReport?.rows?.[ medianIndex ]?.metricValues?.[ 0 ]?.value,
+			10
+		) || 0;
+
+	const reportOptions = {
+		...dates,
+		dimensions: [ 'pagePath' ],
+		metrics: [ 'bounceRate', 'screenPageViews' ],
+		orderby: [
+			{
+				metric: { metricName: 'bounceRate' },
+				desc: true,
+			},
+			{
+				metric: { metricName: 'screenPageViews' },
+				desc: true,
+			},
+		],
+		metricFilters: {
+			screenPageViews: {
+				operation: 'GREATER_THAN_OR_EQUAL',
+				value: { int64Value: medianPageViews },
+			},
+		},
+		limit: 3,
+	};
+
+	const loadedPageViewsReport = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).hasFinishedResolution( 'getReport', [
+			pageViewsReportOptions,
+		] )
+	);
+
+	const pageViewsReportError = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getErrorForSelector( 'getReport', [
+			pageViewsReportOptions,
+		] )
+	);
+
+	const report = useSelect( ( select ) => {
+		if ( loadedPageViewsReport && ! pageViewsReportError ) {
+			return select( MODULES_ANALYTICS_4 ).getReport( reportOptions );
+		}
```

Comment:
Although we don't have an explicit linting rule for this, we do generally avoid the implicit return, i.e. a value-returning function should always return its value via an explicit `return` statement.

With this in mind I'd suggest rewriting this function body as follows - we can return `null` when there's an error as it's no longer conceptually "loading", rather it's in the error state:

```suggestion
		if ( ! loadedPageViewsReport ) {
			return undefined;
		}

		if ( pageViewsReportError ) {
			return null;
		}

		return select( MODULES_ANALYTICS_4 ).getReport( reportOptions );
```

So we can catch this sort of thing automatically in future, I'll create an issue to add the [`consistent-return`](https://eslint.org/docs/latest/rules/consistent-return) ESLint rule to our config.

Reply:
Further to the above, I've created the new issue https://github.com/google/site-kit-wp/issues/7645.